### PR TITLE
change from is_native_array to is_ivy array in get_all_arrays_in_memory

### LINKED ISSUE #10428: [Bug]: ivy.get_all_arrays_in_memory() produces errors
### Bug Explanation

when using get_all_arrays_in_memory function gives below error
numpy: is_native_array: maximum recursion depth exceeded in comparison 
It should return ivy.Array but it returns garbage value

### Steps to Reproduce Bug

ex:- get_all_arrays_in_memory() in python console 

### Environment

ArchLinux ,venv environment

### Ivy Version

v1.1.9

### Backend

- [X] NumPy
- [ ] TensorFlow
- [ ] PyTorch
- [ ] JAX

### Device

CPU

### PR DESCRIPTION
**Pull request by Killua7362**
This fixes #10428

### CODE REVIEW

**Comment by Killua7362** [comment]
Error is not related to my changes can you please look into this

**Comment by hirwa-nshuti** [comment]
Hey @Killua7362 would you add examples highlighting the failures🙂

**Comment by Killua7362** [comment]
I get this error
ivy is in /home/killua/repo directory

>>> from ivy import get_all_arrays_in_memory
>>> get_all_arrays_in_memory()
  File "/home/killua/repo/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/home/killua/repo/ivy/ivy/functional/ivy/general.py", line 155, in is_native_array
    return current_backend(x).is_native_array(x, exclusive=exclusive)
  File "/home/killua/repo/ivy/ivy/backend_handler.py", line 220, in current_backend
    f = _determine_backend_from_args(list(args) + list(kwargs.values()))
  File "/home/killua/repo/ivy/ivy/backend_handler.py", line 94, in _determine_backend_from_args
    lib = _determine_backend_from_args(arg)
  File "/home/killua/repo/ivy/ivy/backend_handler.py", line 87, in _determine_backend_from_args
    lib = _determine_backend_from_args(value)
  File "/home/killua/repo/ivy/ivy/backend_handler.py", line 87, in _determine_backend_from_args
    lib = _determine_backend_from_args(value)
  File "/home/killua/repo/ivy/ivy/backend_handler.py", line 87, in _determine_backend_from_args
    lib = _determine_backend_from_args(value)
  [Previous line repeated 984 more times]
  File "/home/killua/repo/ivy/ivy/backend_handler.py", line 81, in _determine_backend_from_args
    if isinstance(args, ivy.Array):
  File "/usr/lib/python3.10/abc.py", line 119, in __instancecheck__
    return _abc_instancecheck(cls, instance)

During the handling of the above exception, another exception occurred:

[masked_array(data=--,
             mask=True,
       fill_value=1e+20,
            dtype=float64), masked]
>>> 

I tried getting the output of the error which is given below
**numpy: is_native_array: maximum recursion depth exceeded in comparison**

File: ivy/functional/ivy/general.py [hunk 1]
```diff
@@ -2177,15 +2177,12 @@ def stable_pow(
 
 
 @handle_exceptions
-def get_all_arrays_in_memory():
+def get_all_arrays_in_memory() -> ivy.Array:
     """Gets all arrays which are currently alive."""
     all_arrays = list()
     for obj in gc.get_objects():
```
[the remark this answers is not in the document]
Comment (Killua7362) [reply ivy/functional/ivy/general.py:2183]: Thank you I'll look into it

[the remark this answers is not in the document]
Comment (hirwa-nshuti) [reply ivy/functional/ivy/general.py:2183]: Yes when it the backend is `numpy` and `""`

[the remark this answers is not in the document]
Comment (Killua7362) [reply ivy/functional/ivy/general.py:2183]: Will this work 

    for obj in gc.get_objects():
        try:
            if isinstance(o,ivy.Array) and ivy.is_native_array(obj) :
                all_arrays.append(obj)
        except Exception:
            pass

[the remark this answers is not in the document]
Comment (hirwa-nshuti) [reply ivy/functional/ivy/general.py:2183]: No this will throw the same error as first one
What I would recommend is adding another try block for that condition I mentioned and keep the existing one as else part

[the remark this answers is not in the document]
Comment (hirwa-nshuti) [reply ivy/functional/ivy/general.py:2183]:
```python3

if ivy.current_backend_str() == "numpy" or ivy.current_backend_str() == "":
        for obj in gc.get_objects():
            try:
                if ivy.is_ivy_array(obj):
                    all_arrays.append(obj)
            except Exception:
                pass
        return all_arrays
    for obj in gc.get_objects():
        try:
            if ivy.is_native_array(obj):
                all_arrays.append(obj)
        except Exception:
            pass
```
I was thinking of something like this😊

[the remark this answers is not in the document]
Comment (Killua7362) [reply ivy/functional/ivy/general.py:2183]: > ```python
> if ivy.current_backend_str() == "numpy" or ivy.current_backend_str() == "":
>         for obj in gc.get_objects():
>             try:
>                 if ivy.is_ivy_array(obj):
>                     all_arrays.append(obj)
>             except Exception:
>                 pass
>         return all_arrays
>     for obj in gc.get_objects():
>         try:
>             if ivy.is_native_array(obj):
>                 all_arrays.append(obj)
>         except Exception:
>             pass
> ```
> 
> I was thinking of something like thisblush

I mean can't we do it with single iteration?

[the remark this answers is not in the document]
Comment (Killua7362) [reply ivy/functional/ivy/general.py:2183]: > This will return empty list when ivy arrays are stored in memory so for numpy backend and default backend ivy arrays are also stored. If you find a one iteration to handle this case as well it will be nice and may above example is a naive solution which covers both cases

```
def get_all_arrays_in_memory() -> ivy.Array:
    """Gets all arrays which are currently alive."""
    all_arrays = list()
    for obj in gc.get_objects():
        try:
            if ivy.current_backend_str() in ['','numpy']:
                if ivy.is_ivy_array(obj):
                    all_arrays.append(obj)
            else:
                if ivy.is_native_array(obj):
                    all_arrays.append(obj)

        except Exception:
            pass
    return all_arrays

```
You mean like this? do we really need two try_except block?

[the remark this answers is not in the document]
Comment (hirwa-nshuti) [reply ivy/functional/ivy/general.py:2183]: Yea this one can work👍

File: ivy/functional/ivy/general.py [hunk 2]
```diff
@@ -2177,13 +2177,18 @@ def stable_pow(
 
 
 @handle_exceptions
-def get_all_arrays_in_memory():
+def get_all_arrays_in_memory() -> ivy.Array:
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/general.py:2180]: This function returns a list of arrays
```suggestion
def get_all_arrays_in_memory() -> List[ivy.Array, ivy.NativeArray]:
```

Comment (Killua7362) [reply ivy/functional/ivy/general.py:2180]: My bad
 i will fix the code and push it tomorrow

**Review by hirwa-nshuti — APPROVED** [review]
LGTM😊
Thanks for contributing

**Comment by Killua7362** [comment]
> LGTMblush Thanks for contributing

Thank you for taking your time
